### ISSUE TITLE
Question about model training with checkpointing:

### ISSUE THREAD

**Issue by mostafamdy** [huggingface/transformers#31044]
### System Info

I currently download the dataset and split it into training and evaluation sets using `dataset.train_test_split(test_size=0.005, seed=42)` every time I run the code.

My question is: When I resume training from a checkpoint with `trainer.train(resume_from_checkpoint="trained_weights/checkpoint-1000")`, will the model train on the entire dataset again, or will it skip processing the data until it reaches the saved step (step 1000 in this case)?

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

.

### Expected behavior

.

**Comment by RUFFY-369** [comment]
@mostafamdy Have a look at the code from this [line](https://github.com/huggingface/transformers/blob/bdb9106f247fca48a71eb384be25dbbd29b065a8/src/transformers/trainer.py#L2125). So, the `Trainer` class have an arg named `ignore_data_skip`. It governs **"while resuming training, whether or not to skip the first epochs and batches to get to the same training data"**. And its default value is `False`. So, skip processing the data is upto you.

Cheers!
